test/integration/admin/api/member_permissions_controller_test.rb

```diff
@@ -0,0 +1,146 @@
+
+    login! @provider
+  end
+
+  test 'get' do
+    ThreeScale.stubs(master_on_premises?: true)
+    get admin_api_permissions_path(id: @user.id, format: :json)
+    assert_response :success
+  end
+
+  test "PUT: enable 'analytics' section for service 1" do
+    params = { allowed_sections: ['monitoring'], allowed_service_ids: [1] }
+
+    put admin_api_permissions_path(id: @user.id, format: :json), params
+
+    permissions = JSON.parse(@response.body)['permissions']
+    assert_equal ['monitoring'], permissions['allowed_sections']
+    assert_equal [1], permissions['allowed_service_ids']
+  end
+
+  test "PUT: enable 'settings', but keep the same services" do
+    @user.update_attributes({ allowed_sections: ['partners'], allowed_service_ids: [1] })
+    params = { allowed_sections: ['settings'] }
+
+    put admin_api_permissions_path(id: @user.id, format: :json), params
+
+    permissions = JSON.parse(@response.body)['permissions']
+    assert_equal ['settings'], permissions['allowed_sections']
+    assert_equal [1], permissions['allowed_service_ids']
+  end
+
+  test "PUT: enable service '2', but keep the same sections" do
+    @user.update_attributes({ allowed_sections: ['partners'], allowed_service_ids: [1] })
+    params = { allowed_service_ids: ['2'] }
+
+    put admin_api_permissions_path(id: @user.id, format: :json), params
+
+    permissions = JSON.parse(@response.body)['permissions']
+    assert_equal ['partners'], permissions['allowed_sections']
+    assert_equal [2], permissions['allowed_service_ids']
+  end
+
+  test "PUT: enable 'settings' and enable all services" do
+    # allowed_sections%5B%5D=settings&allowed_service_ids%5B%5D
+    params = { allowed_sections: ['settings'], allowed_service_ids: nil }
+
+    put admin_api_permissions_path(id: @user.id, format: :json), params
+
+    permissions = JSON.parse(@response.body)['permissions']
+    assert_equal ['settings'], permissions['allowed_sections']
+    assert_nil permissions['allowed_service_ids']
+  end
+
+  test "PUT: enable 'settings', but disable all services" do
+    # allowed_sections%5B%5D=settings&allowed_service_ids%5B%5D=%5B%5D
+    params = { allowed_sections: ['settings'], allowed_service_ids: ["[]"] }
+
+    put admin_api_permissions_path(id: @user.id, format: :json), params
+
+    permissions = JSON.parse(@response.body)['permissions']
+    assert_equal ['settings'], permissions['allowed_sections']
+    assert_empty permissions['allowed_service_ids']
+  end
+
+  test "updating admin's permissions is not allowed" do
+    @user.update_attribute :role, 'admin'
+    params = { allowed_sections: ['monitoring'], allowed_service_ids: [1] }
+
+    put admin_api_permissions_path(id: @user.id, format: :json), params
+
+    assert_response :forbidden
+    assert_equal '{"status":"Forbidden"}', @response.body
+  end
+
+  test "member user can't update his own permissions" do
+    @user.update_attribute :role, 'member'
+    provider_logout
```

Comment:
Is this necessary? It works for me without it 🤔 
![image](https://user-images.githubusercontent.com/11318903/49213638-03a43480-f3c5-11e8-9039-163fbae74527.png)